frontend-web/webclient/app/ui-components/RichSelect.tsx

```diff
@@ -48,11 +49,15 @@ export function RichSelect<T, K extends keyof T>(props: {
     }, []);
 
     return <ClickableDropdown
-        trigger={
-            <div className={TriggerClass} ref={triggerRef}>
-                <props.RenderSelected element={props.selected} onSelect={doNothing}/>
-                <Icon name="chevronDownLight"/>
-            </div>
+        trigger={props.FullRenderSelected ? 
+                <props.FullRenderSelected element={props.selected} onSelect={doNothing} />
+            :
+                props.RenderSelected ? 
+                    <div className={TriggerClass} ref={triggerRef}>
+                        <props.RenderSelected element={props.selected} onSelect={doNothing}/>
+                        <Icon name="chevronDownLight"/>
+                    </div>
+                : <></>
```

Comment:
[Non-blocking] If I'm reading this correctly, only one of `FullRenderSelected` and `RenderSelected` will be rendered. Can't you just pass the full component down once instead of allowing to different?